src/parse/state/post.js

```diff
@@ -15,4 +17,12 @@ function species (spec) {
   }
 }
 
+function boundary (spec, species) {
+  if (species === PostSpecies.Structured && spec.type.split(';')[0] === 'multipart/form-data') {
```

Comment:
> `spec.type.split(';')[0]`

This seems to be done all over the place. Maybe create a function for it? 

> DISCLAIMER: The following code has not been tested 😁 
```
const parseContentType = str => {
  const [mimeType, ...rest] = str.split(';').map(s => s.trim())
  const params = Object.fromEntries(rest.map(s => s.split('=')))

  return {
    ...params,
    mimeType,
  }
}
```

That way you also get the charset and the boundary as well. :)

Reply:
Yes I agree, it really should be extracted to a helper!
God idea, will implement 👍